lib/rubocop/cop/graphql/field_name.rb

```diff
@@ -31,7 +32,34 @@ def on_send(node)
           field = RuboCop::GraphQL::Field.new(node)
           return if field.name.snake_case?
 
-          add_offense(node)
+          add_offense(node) do |corrector|
+            rename_field_name(corrector, field, node)
+          end
+        end
+
+        private
+
+        def rename_field_name(corrector, field, node)
+          name_field = field.name.to_s
+          new_line = node.source.sub(name_field, underscore(name_field))
+          corrector.replace(node, new_line)
+        end
+
+        # This method was extracted from activesupport in Rails:
+        # https://github.com/rails/rails/blob/8dab534ca81dd32c6a83ac03596a1feb7ddaaca7/activesupport/lib/active_support/inflector/methods.rb#L96
+
+        def underscore(camel_cased_word)
```

Comment:
That's a perfect use case for [refinements](https://docs.ruby-lang.org/en/3.0.0/doc/syntax/refinements_rdoc.html)! We could add a refinement for a string, check if Rails (or activesupport specifically) is defined and, if not, _refine_ the string.

Do you want to try it out? That's completely optional and just for fun, I'd be happy to merge this PR as is

Reply:
Good idea, I haven't had a chance to try that out yet!

I pushed up a commit, not sure if that's how it's best structured, I'm happy to listen to feedback if not.